lib/expect.toit

```diff
@@ -117,11 +117,12 @@ expect_bytes_equal expected/ByteArray actual/ByteArray:
 /** Expects $throw_block to throw an object equal to the $expected value. */
 expect_throw expected [throw_block]:
   try:
-     throw_block.call
-  finally: | is_exception exception |
-    if not is_exception:
+    error := catch --trace=(: expected != it) --unwind=(: expected != it):
+      throw_block.call
+    if expected != error:
       expect false --message="Expected throw, got <null>"
-    if expected != exception.value:
+  finally: | is_exception exception |
```

Comment:
Now we would change the message from "Expected throw, got <null>" to "Expected throw with <...>, but was Expected throw, got <null>".

I think the `finally` clause should disappear.